training/data/dataset_processing_example_blanks.R

```diff
@@ -2,78 +2,39 @@
 
 # Reminder Get test server token
 
-#Load your libraries
+library(dataone)
+library(datapack)
+library(digest)
+library(uuid)
 library(dataone)
 library(arcticdatautils)
 library(EML)
-library(tidyverse)
-
-#set up your nodes (in this case using the arcticdata node)
-cn <- CNode("___________")
-adc <- getMNode(cn,"___________") # the mn
-
-#get the resource map
-rm_original <- "___________"
-
-#make sure you have the most recent resource map in the version chain
-all_versions <- get_all_versions(adc, _______)
-rm <- ______[length(_______)]
-
-#get the package
-pkg <- get_package(adc, rm) # has the resource map, metadata and data PIDs
-doc <- read_eml(getObject(adc, _______)) # the metadata document
-
-#fix up the system metadata
-sysmeta <- getSystemMetadata(adc, __________)
-sysmeta@formatId <- "___________" #set the right formatId
-updateSystemMetadata(adc, ________, sysmeta) #make sure changes are updated
-
-###### Make your changes to the metadata ###########
-
-# adding FAIR information
-doc <- eml_add_publisher(____)
-doc <- eml_add_entity_system(____)
-
-#the below will change depending on what your dataset needs
-#convert other to datatable
-doc <- eml_otherEntity_to_dataTable(doc, ____, validate_eml = F)
-
-#attribute table
-data <- read.csv("_____________")
-att <- EML::shiny_attributes(data = _____)
-write_csv(att$_______, "attribute_tbl.csv") #save your attribute table for later just incase
-
-att_tbl <- read_csv("attribute_tbl.csv")
-att <- EML::shiny_attributes(attributes = att_tbl)
-att <- set_attributes(att$attributes)
 
-#set physicals
-physical <- pid_to_eml_physical(adc, pkg$_____)
+d1c <- D1Client("____________", "____________")
+packageId <- "____________"
+dp <- getDataPackage(____________, identifier=____________, lazyLoad=TRUE, quiet=FALSE)
 
-dataTable <- eml$dataTable(entityName = "_____",
-                           entityDescription = "_________",
-                           physical = ______,
-                           attributeList = _____)
+#get metadata id
+metadataId <-____________(dp, name="sysmeta@formatId", value="https://eml.ecoinformatics.org/eml-2.2.0")
 
-doc$dataset$dataTable <- _______
+#edit the metadata
+doc <- ____________(getObject(____________, ____________))
 
-###### Check your changes ###########
+#add the publisher info
+doc <- eml_add_publisher(doc)
+doc <- eml_add_entity_system(doc)
 
-#validate
-eml_validate(______) # errors? go back and change the above until it passes as TRUE
-eml_path <- "example_eml.xml" # where your file will be saved
-write_eml(______, _______) # save your metadata
+#format award info
+doc$dataset$project <- ____________("____________")
 
-###### Publish your changes ###########
+#check and save the metadata
+____________(____________)
+eml_path <- paste0(str_replace_all(word(doc$dataset$title, end = 6), " ", "_"), ".xml")
```

Comment:
yikes, that is pretty opaque. maybe we should isolate the helper function from arcticdatautils here and export it

Reply:
Agreed! Been meaning to do that and overhaul the entire arcticdatutils package. Been doing some work here on and off just to [evaluate what is in the package](https://docs.google.com/spreadsheets/d/1yxBqDZsm1PbGzaPUa2NrPxqsgvP1VMd1eq1ksxkrD40/edit#gid=1148653955). I think I'm now at a point where we can sit down and  go through this together

Reply:
yeah definitely this is a nice looking sheet. Sysmeta helpers in particular are crazy useful for us (`set_access`,  `get_all_versions`) and don't really have a great home in other packages